sys/net/gnrc/Makefile.dep

```diff
@@ -10,10 +10,6 @@ ifneq (,$(filter sock_async,$(USEMODULE)))
   USEMODULE += gnrc_sock_async
 endif
 
-ifneq (,$(filter sock_peek,$(USEMODULE)))
-  $(error GNRC does not support sock_peek)
-endif
-
```

Comment:
Unless you plan to add support for UDP and IP with this PR as well we should keep the message and only remove it for TCP.

Reply:
How would I express that in modules, given there is no `sock_peek_tcp`, `sock_peek_udp` and `sock_peek_ip`?

Reply:
> How would I express that in modules, given there is no `sock_peek_tcp`, `sock_peek_udp` and `sock_peek_ip`?

Mh, we could check for the use of `sock_udp` and `sock_ip` and make the error a warning instead. This + an extra note for the documentation of SOCK_AUX_PEEK (once it is merged) should be ok in my opinion.